### DISCUSSION TITLE
How to remove Card box shadow that act as border

### DISCUSSION THREAD

**Discussion by thewebartisan7** [radix-ui/themes#450] [Q&A]
Below is the code of a `Card` that has a `Table`, and I am using `Inset` to remove left/right space for `Table`.

The only problem is that the default `Card` box shadow can't be easily removed, and with variant ghost also all padding is lost and `Inset` doesn't works properly.

The only workaround was to remove the box shadow via inline style, and it was pretty hard to find this variable that act as border.

I tried with a `Box` but `Inset` with `Box` doesn't seems to works fine. I want the `Table` to start from edge of screen but has inner padding, something like in your demo where there is `Table` inside `Dialog`.

Code with workaround:

```js
<Card size="4" style={{ '--base-card-surface-box-shadow': 'none' } as any}>
                <Heading as="h2" size="6" mb="5">
                    Table
                </Heading>

                <Inset clip="padding-box" side="x">
                    <Table.Root>
                        <Table.Header>
                            <Table.Row>
                                <Table.ColumnHeaderCell>ID</Table.ColumnHeaderCell>
                                <Table.ColumnHeaderCell>Name</Table.ColumnHeaderCell>
                            </Table.Row>
                        </Table.Header>

                        <Table.Body>
                             <Table.Row>
                                    <Table.RowHeaderCell width="150px">1</Table.RowHeaderCell>
                                    <Table.Cell>John</Table.Cell>
                                </Table.Row>
                        </Table.Body>
                    </Table.Root>
                </Inset>
            </Card>
```

Is there a better way?

Thanks

**Comment by kevinmitch14** [comment]
What if you use `variant="ghost"` + `className="m-0"` or `style={{margin:0}}`?

**Reply by thewebartisan7** [reply]
Unfortunately it's not working. I tried with ghost but `Inset` then doesn't works.

**Comment by vladmoroz** [comment]
@thewebartisan7 what you did makes sense to me, changes to most variables are usually going to be documented in the changelog.

Why do you need to use the Card though? If you can get away with the Box and some padding, I'd just go with equivalent negative margin on the Table to achieve what Inset achieves.

**Comment by thewebartisan7** [comment]
I don't need to use the `Card` , it's just the only one that worked with `Inset` when I tried.

For example below with `Box`, without `Inset`, and with negative margin on `Table` doesn't works, I mean the only spacing is of cells.
Since `Table` has not padding I am unsure how to add space to it so that row border are from edge to edge.

```ts
<Box px="6" my="6">
                <Heading as="h2" size="6" mb="5">
                    Users
                </Heading>

                <Table.Root mx="-6">
                     <Table.Header>
                            <Table.Row>
                                <Table.ColumnHeaderCell>ID</Table.ColumnHeaderCell>
                                <Table.ColumnHeaderCell>Name</Table.ColumnHeaderCell>
                            </Table.Row>
                        </Table.Header>

                        <Table.Body>
                             <Table.Row>
                                    <Table.RowHeaderCell width="150px">1</Table.RowHeaderCell>
                                    <Table.Cell>John</Table.Cell>
                                </Table.Row>
                        </Table.Body>
                </Table.Root>
            </Box>
```

**Reply by vladmoroz** [reply]
Do you have a reference design for what you are trying to achieve?

Although again, your original implementation is fine, just want to see if there's a more native way of doing that.

**Reply by thewebartisan7** [reply]
This with Card + Inset and inline variable overrides, which is how it should looks like:

<img width="1644" alt="Screenshot 2024-04-10 at 21 23 35" src="https://github.com/radix-ui/themes/assets/57477934/b8ee8bd9-311f-4b71-ab17-67a6ba1b73c1">

While this is with Box + Table and negative margin:

<img width="1659" alt="Screenshot 2024-04-10 at 21 11 13" src="https://github.com/radix-ui/themes/assets/57477934/6693a3e5-5d3d-4a20-89c6-d4555e1331bd">

Yes I can stay with inline styles, but as you said would be good to have a native way of doing this.

**Reply by vladmoroz** [reply]
Ah now it's clear, the design makes a lot of sense to me. We'll keep this use case in mind as we improve the related components.

I'd say that at the moment, the best way to achieve that would be to use the `pl` and `pr` props to override the table cell padding ([see docs](https://www.radix-ui.com/themes/docs/components/table#cell)) and to set an equivalent negative margin on the table root or its container element. If you use this table style a lot, you could create your own table-related components that would have predefine these values so you don't override padding and margin every time you need to display a table.

Wonder if an API where setting a default `pl` and `pr` on Table.Root element itself would be more intuitive.

**Reply by thewebartisan7** [reply]
Ah yes adding padding on cell could work, will try. Thanks!

> Wonder if an API where setting a default pl and pr on Table.Root element itself would be more intuitive.

Yes that would ideal so you can just set it once on Root.

Another question about `Table`. I see that you wrap table inside `ScrollArea`, but can't see how to customize it. For example how to make the table content scrollable?

The only way seems to see the height of `Table.Root` via inline style since `Table.Root` has not height prop, example:

```js
<Table.Root style={{ height: '100dvh' }}>
```

Result:

https://gyazo.com/dd88223d33c9f80dbc11a26f8158cc21

![dd88223d33c9f80dbc11a26f8158cc21](https://github.com/radix-ui/themes/assets/57477934/69791ce0-6f90-4e68-b97e-9a6fee07bc11)

And would be nice to make scrollable only Table.Body so that TableHeader is sticky on top.. but this can be archived also with custom style.

**Reply by vladmoroz** [reply]
> Another question about Table. I see that you wrap table inside ScrollArea, but can't see how to customize it. For example how to make the table content scrollable?

This is primarily for automatic horizontal scroll when table contents don't fit into the table’s container, but should work with vertical scroll if you constrain the height of the table. Inline height style is totally fine

> And would be nice to make scrollable only Table.Body so that TableHeader is sticky on top.. but this can be archived also with custom style.

Yep, at the moment this would have to be done with custom styles applied to the table headers
